### PR TITLE
rollup table for rules analysis

### PR DESCRIPTION
**Pull request by emilia-friedberg**
## WHAT
Update the rules analysis frontend to rollup the new data that is passed to it by API.

## WHY
This is an improvement on the current view that was requested by the curriculum team.

## HOW
Use `react-table`'s extremely handy pivot table functionality to roll up and aggregate data by API.

### Screenshots
Screen Shot 2021-05-12 at 1 03 40 PM
Screen Shot 2021-05-12 at 1 03 53 PM

### Notion Card Links
https://www.notion.so/quill/Frontend-Rules-Report-Rollup-Individual-Rules-to-Summary-of-API-01a2c161cd1b4ee1a81c7482f6acd3a3?d=f7b91590-b001-4a8b-9a80-b60f61d0219a#034a9693d29d4146a880edd6cb3396d1

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  Manually tested
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES
Design Review: If applicable, have you compared the coded design to the mockups? | YES

### CODE REVIEW

**Review by anathomical — APPROVED** [review]

**Comment by happythenewsad** [comment]
@emilia-friedberg This diff contains commits from 4 different people. I'm not sure which of the 42 files to review.

**Comment by emilia-friedberg** [comment]
> @emilia-friedberg This diff contains commits from 4 different people. I'm not sure which of the 42 files to review.

Yeah, great question. It's because this branch is being compared to @anathomical's and I think I have a more recent version of develop pulled. I'll flag the files that are actually mine here.

File: services/QuillLMS/client/app/bundles/Staff/components/comprehension/rulesAnalysis/rulesAnalysis.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { useQuery } from 'react-query';
 import { firstBy } from "thenby";
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/components/comprehension/rulesAnalysis/rulesAnalysis.tsx:4]: This file

File: services/QuillLMS/client/app/bundles/Staff/styles/rules_analysis.scss [hunk 2]
```diff
@@ -11,6 +11,12 @@
     margin-right: 40px;
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Staff/styles/rules_analysis.scss:11]: And this file.

**Review by happythenewsad — APPROVED** [review]

**Pull request merged**